src/messagetypes/__init__.py

```diff
@@ -7,7 +7,7 @@
 import paths
 
 class MsgBase(object):
-    def encode(self):
+    def encode(self, data=None):
         self.data = {"": lower(type(self).__name__)}
```

Comment:
what you could do is to make this a default (`if data is None:`), otherwise assign `data` to `self.data`